client/my-sites/checkout/checkout/wechat-payment-box.jsx

```diff
@@ -0,0 +1,258 @@
+							/>
+						</div>
+					</div>
+
+					{ children }
+
+					<TermsOfService
+						hasRenewableSubscription={ cartValues.cartItems.hasRenewableSubscription( cart ) }
+					/>
+
+					<div className="checkout__payment-box-actions">
+						<div className="checkout__pay-button">
+							<button
+								type="submit"
+								className="checkout__button-pay button is-primary "
+								disabled={ this.state.createFetching }
+							>
+								{ translate( 'Pay %(price)s with WeChat Pay', {
+									args: { price: cart.total_cost_display },
+								} ) }
+							</button>
+							<SubscriptionText cart={ cart } />
+						</div>
+
+						{ showPaymentChatButton && (
+							<PaymentChatButton paymentType={ paymentType } cart={ cart } />
+						) }
+					</div>
+				</form>
+
+				<CartCoupon cart={ cart } />
+
+				<CartToggle />
+			</React.Fragment>
+		);
+	}
+}
+
+export const requestId = cart =>
+	`my-sites/checkout/checkout/wechat-payment-box/${ get( cart, 'cart_key', '0' ) }`;
+
+export const getWeChatTransactionOrderDetails = ( { cart, domainDetails, payment } ) => {
+	return requestHttpData(
+		requestId( cart ),
+		http( {
+			path: '/me/transactions',
+			apiVersion: '1.1',
+			method: 'POST',
+			body: { cart, domain_details: domainDetails, payment },
+		} ),
+		{
+			fromApi: () => ( { order_id, redirect_url } ) => [
+				[ requestId( cart ), { orderId: order_id, redirectUrl: redirect_url } ],
+			],
+			freshness: -Infinity,
+		}
+	);
+};
+
+export default connect(
+	( state, { cart } ) => {
+		const wechatOrder = getHttpData( requestId( cart ) ) || {};
+
+		return {
+			weChatOrderData: get( wechatOrder, 'data', {} ),
+			isRequestingWeChatOrderData: 'pending' === wechatOrder.state,
+			isWeChatOrderDataError: 'failure' === wechatOrder.state,
+			configAge: Date.now() - wechatOrder.lastUpdated,
+		};
+	},
+	( dispatch, { cart, transaction, selectedSite, redirectTo } ) => ( {
+		fetchWeChatOrder: name => {
+			const origin = getLocationOrigin( location );
+
+			const slug = get( selectedSite, 'slug', 'no-site' );
+
+			const payment = {
+				name,
+				payment_method: paymentMethodClassName( 'wechat' ),
+				success_url: origin + redirectTo(),
+				cancel_url: `${ origin }/checkout/${ slug }`,
+			};
+
+			dispatch( infoNoticeAction( translate( 'Setting up your WeChat Pay payment' ) ) );
+
+			dispatch( recordTracksEvent( 'calypso_checkout_with_redirect_wechat' ) );
+			dispatch( recordGoogleEvent( 'Upgrades', 'Clicked Checkout With Wechat Payment Button' ) );
+
+			getWeChatTransactionOrderDetails( {
+				cart,
+				domainDetails: transaction.domainDetails,
+				payment,
+			} );
+		},
+		// We should probably extend `update()` in client/state/data-layer/http-data.js
+		// to set state back to 'uninitialized'
+		resetWeChatOrder: () =>
+			window.httpData.set( requestId( cart ), {
+				state: 'uninitialized',
+				data: {},
+				error: undefined,
+			} ),
+		errorNotice: errorNoticeAction,
+		infoNotice: infoNoticeAction,
```

Comment:
We have to manually bind these to `dispatch` as well. So we could create new functions so:

```
		showErrorNotice: ( error, options ) => dispatch( errorNotice( error, options ) ),
		showInfoNotice: ( info, options ) => dispatch( infoNotice( info, options ) ),
```